test/gtest/pdist.hpp

```diff
@@ -0,0 +1,228 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include <miopen/miopen.h>
+#include <miopen/pdist.hpp>
+#include <gtest/gtest.h>
+
+#include "get_handle.hpp"
+#include "random.hpp"
+#include "tensor_holder.hpp"
+#include "verify.hpp"
+#include "cpu_pdist.hpp"
+
+struct PdistTestCase
+{
+    std::vector<size_t> dims;
+    double p;
+
+    friend std::ostream& operator<<(std::ostream& os, const PdistTestCase& tc)
+    {
+        os << "dims: ( ";
+        for(auto d : tc.dims)
+        {
+            os << d << " ";
+        }
+        os << ") ";
+        os << "p: " << tc.p;
+
+        return os;
+    }
+
+    std::vector<size_t> GetDims() const { return dims; }
+    double GetP() const { return p; }
+
+    PdistTestCase() {}
+
+    PdistTestCase(std::vector<size_t> dims_, double p_) : dims(dims_), p(p_) {}
+};
+
+inline std::vector<PdistTestCase> PdistTestConfigs()
+{
+    return {
+        // TODO: How to test when expected output=tensor([])?
+        // PdistTestCase({1, 1}, 0.0),
+        // PdistTestCase({1, 1}, 1.0),
+        // PdistTestCase({1, 1}, 2.0),
+        // PdistTestCase({1, 1}, 5.0),
+        // PdistTestCase({1, 1}, HUGE_VAL),
```

Comment:
remove comments

Reply:
Those tests are for cases when output is empty tensor: `tensor([])`. I'm fixing my op for this cases. I'll update this asap

Reply:
pls check my discussion [here](https://github.com/moreh-dev/MIOpen/pull/62#discussion_r1855749335)

Reply:
solved: throw when input/output is empty tensor